pkg/cli/clients/management_mocks.go

```diff
@@ -71,3 +71,18 @@ type resourceGroupClient interface {
 	Get(ctx context.Context, planeName string, resourceGroupName string, options *ucpv20231001.ResourceGroupsClientGetOptions) (ucpv20231001.ResourceGroupsClientGetResponse, error)
 	NewListPager(planeName string, options *ucpv20231001.ResourceGroupsClientListOptions) *runtime.Pager[ucpv20231001.ResourceGroupsClientListResponse]
 }
+
+// resourceProviderClient is an interface for mocking the generated SDK client for resource providers.
+type resourceProviderClient interface {
+	BeginCreateOrUpdate(ctx context.Context, planeName string, resourceProviderName string, resource ucpv20231001.ResourceProviderResource, options *ucpv20231001.ResourceProvidersClientBeginCreateOrUpdateOptions) (*runtime.Poller[ucpv20231001.ResourceProvidersClientCreateOrUpdateResponse], error)
+	BeginDelete(ctx context.Context, planeName string, resourceProviderName string, options *ucpv20231001.ResourceProvidersClientBeginDeleteOptions) (*runtime.Poller[ucpv20231001.ResourceProvidersClientDeleteResponse], error)
+	Get(ctx context.Context, planeName string, resourceProviderName string, options *ucpv20231001.ResourceProvidersClientGetOptions) (ucpv20231001.ResourceProvidersClientGetResponse, error)
+	NewListPager(planeName string, options *ucpv20231001.ResourceProvidersClientListOptions) *runtime.Pager[ucpv20231001.ResourceProvidersClientListResponse]
+	GetProviderSummary(ctx context.Context, planeName string, resourceProviderName string, options *ucpv20231001.ResourceProvidersClientGetProviderSummaryOptions) (ucpv20231001.ResourceProvidersClientGetProviderSummaryResponse, error)
+	NewListProviderSummariesPager(planeName string, options *ucpv20231001.ResourceProvidersClientListProviderSummariesOptions) *runtime.Pager[ucpv20231001.ResourceProvidersClientListProviderSummariesResponse]
+}
+
+// resourceProviderClient is an interface for mocking the generated SDK client for resource types.
```

Comment:
```suggestion
// resourceTypeClient is an interface for mocking the generated SDK client for resource types.
```

Reply:
Thanks for catching this. I'm just about to hit send on another PR that will touch this file, so I'll fix it then if that's OK.